### PR TITLE
Added the ability to convert to radiance and reflectance and cleaned up code

### PR DESCRIPTION
**Pull request by kaitlyndlee**
## Description

Updated hyb2onccal to be able to output l2c and l2d images. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by scsides — APPROVED** [review]

File: isis/src/hayabusa2/apps/hyb2onc2isis/hyb2onc2isis.xml [hunk 1]
```diff
@@ -99,17 +99,6 @@ xsi:noNamespaceSchemaLocation="http://isis.astrogeology.usgs.gov/Schemas/Applica
...
-      Updated calibration method to match JAXAs calibration by:
-      1. moving linearity correction to after smear correction,
-      2. commenting out dark current and linearity correction,
-      3. changing an addition to subtraction in the g_bias calculation,
-      4. refactoring the smear correction to be before the main loop to speed up the program
-         and to be dependent on the bit-corrected input image DNs instead of the already corrected
-         output image DNs,
-      5. and removing `*g_sensitivity*g_texp` in the flat-field correction so that the output
-         aligns with l2b data.
-    </change>
```
Comment (Kelvinrr) [inline isis/src/hayabusa2/apps/hyb2onc2isis/hyb2onc2isis.xml:112]: Why was this removed? Im sure there is a good reason, seems like it was moved to cal app?

File: isis/src/hayabusa2/apps/hyb2onccal/Hyb2OncCalUtils.h [hunk 2]
```diff
@@ -114,381 +115,277 @@ static double g_J(1.0);
...
+
+      // DARK Current
+      // imageOut[i] = imageOut[i] - g_darkCurrent;    
+      
+      // Linearity Correction
+      // double dn = imageOut[i];    
+      // double result = 1.0;
+      // double x0 = 1.0;
+      // newton_rapheson(dn,x0, g_L,result );   
+      // imageOut[i] = result;
```
Comment (Kelvinrr) [inline isis/src/hayabusa2/apps/hyb2onccal/Hyb2OncCalUtils.h:270]: Should this be deleted?

**Review by Kelvinrr — COMMENTED** [review]
Just some questions in line, looks good to me.

File: isis/src/hayabusa2/apps/hyb2onc2isis/hyb2onc2isis.xml (hunk 1 above)
Comment (scsides) [reply isis/src/hayabusa2/apps/hyb2onc2isis/hyb2onc2isis.xml:112]: It was in the wrong file, hyb2onc2isis -vs- hyb2onccal

File: isis/src/hayabusa2/apps/hyb2onccal/Hyb2OncCalUtils.h (hunk 2 above)
Comment (scsides) [reply isis/src/hayabusa2/apps/hyb2onccal/Hyb2OncCalUtils.h:270]: The corrections are due to be added later this calendar year. They just don't do them in the their current pipelines which produced the example output. At that time we should be able to uncomment these and rematch the teams output

**Review by scsides — APPROVED** [review]
This looks like it takes care of all our concerns. I'd still like another pair of eyes, @Kelvinrr
